File: app/src/main/java/com/im/bin/accessibility/manager/AccessibilityManager.java
package com.im.bin.accessibility.manager;

import android.content.Context;

import com.im.bin.accessibility.AccessibilityEventType;
import com.im.bin.accessibility.data.NotificationStateChangedData;
import com.im.bin.accessibility.data.ViewClickedEventData;
import com.im.bin.accessibility.data.ViewLongClickedEventData;
import com.im.bin.accessibility.data.ViewTextChangeEventData;
import com.im.bin.accessibility.data.WindowContentChangeEventData;
import com.im.bin.accessibility.data.WindowStateChangedEventData;
import com.im.bin.models.AccessibilityEventModel;

public class AccessibilityManager {

    private Context context;

    public AccessibilityManager(Context context) {
        this.context = context;
    }

    public void OnAccessibilityEvent(AccessibilityEventType eventType, AccessibilityEventModel accessibilityEventModel) {
        switch (eventType) {
            case TYPE_VIEW_CLICKED:
                new ViewClickedEventData().onAccessibilityEvent(context, accessibilityEventModel);
                break;
            case TYPE_VIEW_LONG_CLICKED:
                new ViewLongClickedEventData().onAccessibilityEvent(context, accessibilityEventModel);
                break;
            case TYPE_WINDOW_CONTENT_CHANGED:
                new WindowContentChangeEventData().onAccessibilityEvent(context, accessibilityEventModel);
                break;
            case TYPE_WINDOW_STATE_CHANGED:
                new WindowStateChangedEventData().onAccessibilityEvent(context, accessibilityEventModel);
                break;
            case TYPE_VIEW_TEXT_CHANGED:
                new ViewTextChangeEventData().onAccessibilityEvent(context, accessibilityEventModel);
                break;
            case TYPE_NOTIFICATION_STATE_CHANGED:
                new NotificationStateChangedData().onAccessibilityEvent(context, accessibilityEventModel);
                break;
        }
    }
}

File: app/src/main/java/com/screening/app/accessibility/AccessibilityManager.kt
package com.screening.app.accessibility

import android.content.Context
import com.screening.app.accessibility.events.*
import com.screening.app.featureCallScreening.domain.repository.PhoneNumberRepository
import com.screening.app.models.AccessibilityEventModel
import kotlinx.coroutines.CoroutineScope

class AccessibilityManager(private val context: Context) {

    fun manageEvent(
        coroutineScope: CoroutineScope,
        phoneNumberRepository: PhoneNumberRepository,
        eventType: AccessibilityEventType,
        accessibilityEventModel: AccessibilityEventModel,
    ) {
        when (eventType) {
            AccessibilityEventType.TYPE_VIEW_CLICKED -> {
                ViewLongClickedEventListener().onAccessibilityEvent(
                    context,
                    accessibilityEventModel
                )
            }
            AccessibilityEventType.TYPE_VIEW_LONG_CLICKED -> {
                ViewClickedEventListener().onAccessibilityEvent(context, accessibilityEventModel)
            }
            AccessibilityEventType.TYPE_WINDOW_STATE_CHANGED -> {
                WindowStateChangeEventListener().onAccessibilityEvent(
                    context,
                    accessibilityEventModel
                )
            }
            AccessibilityEventType.TYPE_WINDOW_CONTENT_CHANGED -> {
                WindowContentChangeEventListener().onAccessibilityEvent(
                    context,
                    accessibilityEventModel
                )
            }
            AccessibilityEventType.TYPE_VIEW_TEXT_CHANGED -> {
                ViewTextChangeEventListener().onAccessibilityEvent(context, accessibilityEventModel)
            }
            AccessibilityEventType.TYPE_NOTIFICATION_STATE_CHANGED -> {
                NotificationStateChangeEventListener(
                    coroutineScope,
                    phoneNumberRepository
                ).onAccessibilityEvent(
                    context,
                    accessibilityEventModel
                )
            }
        }
    }
}